### PR TITLE
Update ASR from LFortran

### PR DESCRIPTION
**Pull request by certik**

### CODE REVIEW

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -1162,39 +1165,69 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
-        this->visit_expr_wrapper(x.m_arg, true);
-        llvm::Value *str = tmp;
-        tmp = lfortran_str_copy(str, idx, idx);
-    }
-
-    void visit_StringSection(const ASR::StringSection_t& x) {
-        if (x.m_value) {
-            this->visit_expr_wrapper(x.m_value, true);
-            return;
+        if (is_a<ASR::Character_t>(*x.m_type)
```
Comment (certik) [inline src/libasr/codegen/asr_to_llvm.cpp:1168]: This part can be removed once LFortran is updated to use the new `StringItem`.

**Comment by certik** [comment]
Also I had to add back some UnaryMinus handling, until LFortran is updated. LPython also needs to be updated to fully remove UnaryMinus.

Otherwise I think this is good.

**Comment by certik** [comment]
Corresponding LFortran sync MR: https://gitlab.com/lfortran/lfortran/-/merge_requests/1764

**Comment by certik** [comment]
@namannimmo10, @Smit-create let me know if you see anything missing or any changed reverted by a mistake. I did my best to merge additions from both LPython and LFortran.

**Comment by certik** [comment]
I am going to go ahead and merge this, so that ASR is now synchronized with LFortran. Then we can submit PRs to both if needed.

File: src/libasr/codegen/asr_to_llvm.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/codegen/asr_to_llvm.cpp:1168]: Fixed in #550.

**Comment by namannimmo10** [comment]
It seems like while making changes to the `libasr` (which LFortran and LPython share,) we should submit PRs to both the projects. Otherwise, we're repeating the steps.

**Comment by namannimmo10** [comment]
> LPython also needs to be updated to fully remove UnaryMinus.

I don't see the occurrence of `UnaryMinus` in the current `main` after it is synchronized. I added that node in #536 but then split that based on different types in #540. 

I guess we just need to update the frontend in LFortran to support `IntegerUnaryMinus`, `RealUnaryMinus`, ... etc.

**Comment by certik** [comment]
I mean UnaryOp, we need to remove UnaryOp.

**Comment by certik** [comment]
Yes, I think when modifying libasr, it's good to synchronize it soon and submit a PR to both projects. It's not a big deal for small changes, but for large refactoring changes we should do it, otherwise it takes me a lot of time to synchronize later.

File: src/libasr/asr_utils.h [hunk 2]
```diff
@@ -289,57 +245,7 @@ static inline std::string boolop_to_str(const ASR::boolopType t) {
...
-        case ASR::exprType::TupleConstant: // Drop through
-        case ASR::exprType::DictConstant: // Drop through
-        case ASR::exprType::SetConstant: // Drop through
-        case ASR::exprType::ListConstant: // Drop through
-        case ASR::exprType::StringConstant:{ // For all Constants
-            return f;
-        }
-        default : throw LFortranException("Not implemented");
-    }
+    return ASR::expr_value0(f);
```
Comment (namannimmo10) [inline src/libasr/asr_utils.h:248]: Happy about this change. :)

Comment (certik) [reply src/libasr/asr_utils.h:248]: I figured you would be. :) It's been bothering me too, so I finally fixed it.

**Pull request merged**